From 91ec92c3a44a59e4935e4accb1b05964c74227d0 Mon Sep 17 00:00:00 2001
From: Bharat Bhushan <Bharat.Bhushan@freescale.com>
Date: Fri, 27 Nov 2015 03:27:14 +0530
Subject: [PATCH 07/18] vfio fsl-mc: Disable device-reset and Misc changes

VFIO_DEVICE_RESET can not be enabled now as this leads to interrupt
configuration cleanup in MC. This will be enabled during mc-portal
emulation.

Also fix to use FSL_MC device type

Signed-off-by: Bharat Bhushan <Bharat.Bhushan@freescale.com>
---
 hw/vfio/fsl_mc.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/hw/vfio/fsl_mc.c b/hw/vfio/fsl_mc.c
index 9675abb..e1841f5 100644
--- a/hw/vfio/fsl_mc.c
+++ b/hw/vfio/fsl_mc.c
@@ -75,7 +75,7 @@ static int vfio_populate_device(VFIODevice *vbasedev)
     VFIOFslmcDevice *vdev =
         container_of(vbasedev, VFIOFslmcDevice, vbasedev);
 
-    if (!(vbasedev->flags & VFIO_DEVICE_FLAGS_PLATFORM)) {
+    if (!(vbasedev->flags & VFIO_DEVICE_FLAGS_FSL_MC)) {
         error_report("vfio: Um, this isn't a fsl_mc device");
         return ret;
     }
@@ -436,7 +436,11 @@ static void vfio_fsl_mc_class_init(ObjectClass *klass, void *data)
     DeviceClass *dc = DEVICE_CLASS(klass);
     FslMcDeviceClass *mcdc = FSL_MC_DEVICE_CLASS(klass);
 
-    dc->reset = vfio_fsl_mc_reset;
+    /* Reset is called after _initfn() and we can not allow reset after
+     * _initfn() as interrupts are setup in _initfn() but a reset to DPRC
+     * will cleanup interrupt configuration in MC.
+     */
+//    dc->reset = vfio_fsl_mc_reset;
     dc->props = vfio_fsl_mc_dev_properties;
     dc->vmsd = &vfio_fsl_mc_vmstate;
     dc->desc = "VFIO-based fsl_mc device assignment";
-- 
1.8.3.2

